### ISSUE TITLE
no-extra-parents typescript .tsx not working right on function calls with generics

### ISSUE THREAD

**Issue by Busin-Riccardo** [eslint/eslint#13879]
**Tell us about your environment**

* **ESLint Version:**
v7.13.0
* **Node Version:**
v12.13.1
* **npm Version:**
v6.14.7

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  plugins: ['react-hooks'],
  parserOptions: {
    ecmaVersion: 2020, 
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true
    },
    project: './tsconfig.json'
  },
  settings: {
    react: {
      version: 'detect'
    }
  },
  extends: 'standard-with-typescript',
  rules: {
    'max-len': [2, {
      ignoreComments: true,
      code: 100
    }],
    '@typescript-eslint/promise-function-async': 'off',
    'react/prop-types': 'off',
    'react-hooks/rules-of-hooks': 'error',
    'no-prototype-builtins': 'off',
    'react-hooks/exhaustive-deps': [
      'warn',
      {
        additionalHooks: 'useRecoilCallback'
      }
    ]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```tsx
memo<{
    item: object
    actions?: Array<(item: object) => React.ReactNode>
  }>(({ item, actions }) => {
    ^-- This is the bracket eslint is telling me isn't needed 
        return <span>{item}</span>
      })
```

```bash
Actually im using Eslint Intellj Plugin.  I think it's calling eslint --fix <filename>
```

**What did you expect to happen?**
Nothing

**What actually happened? Please include the actual, raw output from ESLint.**
Eslint says that first bracket at line 4 isn't needed. This happens only when one of the type properties is a function
For example this works as expected:
```tsx
memo<{
  item: object
  actions?: Array<string> 
}>(({ item, actions }) => {

  return <span>{item}</span>
})
```

**Are you willing to submit a pull request to fix this bug?**
Not really

**Comment by mdjermanovic** [comment]
Hi @Busin-Riccardo, thanks for the issue!

 > ```js
>   extends: 'standard-with-typescript',
> ```

`eslint-config-standard-with-typescript` sets `parser: '@typescript-eslint/parser'` and also enables the core `no-extra-parens` rule which doesn't support typescript code.

You can open an issue in https://github.com/standard/eslint-config-standard-with-typescript repository to update the config to use [@typescript-eslint/no-extra-parens](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-extra-parens.md) rule instead of the core `no-extra-parens` rule, or change it in your config file:

```js
"no-extra-parens": "off",
"@typescript-eslint/no-extra-parens": ["error", "functions"]
```
